pkg/sidecar/fanout/fanout_handler.go

```diff
@@ -83,6 +86,13 @@ func NewHandler(logger *zap.Logger, config Config) *Handler {
 
 func createReceiverFunction(f *Handler) func(provisioners.ChannelReference, *provisioners.Message) error {
 	return func(_ provisioners.ChannelReference, m *provisioners.Message) error {
+		if f.config.AsyncHandler {
```

Comment:
Unit test.

Reply:
Modified existing unit test by adding one test case where AsyncHandler is set true. Let me know  if you think it is sufficient.

Reply:
I don't think anything has been pushed.